akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala

```diff
@@ -36,13 +36,24 @@ import com.typesafe.config.Config
     val recoveryEventTimeout: FiniteDuration =
       journalConfig.getDuration("recovery-event-timeout", TimeUnit.MILLISECONDS).millis
 
+    // Long to allow retention algorithms by time, size or count
```

Comment:
Have this idea in another branch to push for discussion/review after.

Reply:
size will probably be difficult because that would require serialization, and that isn't done at this layer

Reply:
Size of snapshot, not individual messages.